python/cudf/cudf/core/dataframe.py

```diff
@@ -3429,13 +3424,23 @@ def rename(
                     "Implicit conversion of index to "
                     "mixed type is not yet supported."
                 )
-            out = DataFrame(
-                index=self.index.replace(
-                    to_replace=list(index.keys()),
-                    replacement=list(index.values()),
+
+            if level and isinstance(self.index, cudf.core.multiindex.MultiIndex):
+                out = DataFrame(
+                    index=self.index._source_data[level].replace(
```

Comment:
Because we're reaching directly into `_source_data` here we need to be careful to handle the cases where there's duplicate named columns so the names in `self.index.names` doesn't match the names in `self.index._source_data`.

This also looks like it only handles a level name while it should also allow an integer representing the index of the MultiIndex level.